src/Model/Cli/BaseCliCommand.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Waffle\Model\Cli;
+
+use Symfony\Component\Process\Process;
+use Exception;
+use Waffle\Traits\ConfigTrait;
+use Waffle\Model\Config\ProjectConfig;
+
+class BaseCliCommand
+{
+    use ConfigTrait;
+    
+    /**
+     * @var string[]
+     */
+    private $args = [];
+    
+    /**
+     * @var Process
+     */
+    private $process;
+    
+    /**
+     * A reference to the project config.
+     *
+     * @var ProjectConfig
+     */
+    protected $config;
+    
+    /**
+     * Constructor
+     *
+     * @param string[] The Arguments.
+     *
+     * @throws Exception
+     */
+    public function __construct(array $args)
+    {
+        if (empty($args)) {
+            throw new Exception('Invalid Arguments: You must pass at least one argument.');
+        }
+        
+        $this->config = $this->getConfig();
+        
+        // @todo: check for config prefix
+        if (!empty($this->config->getCommandPrefix())) {
+            array_unshift($args, $this->config->getCommandPrefix());
+        }
+        
+        $this->process = new Process($args);
+        $this->process->setTimeout($this->config->getTimeout());
```

Comment:
Similar question here. How do you see this getting used?

Reply:
I had a timeout parameter in the update tool but I was needing to pass it explicitly every time to each process I created. This instead lets you define a timeout globally instead for projects that need it. Some commands like `composer update` can take longer than the default timeout amount.

Reply:
Got it. Looks like you put in a default already, so this looks great.